make_tileset.py

```diff
@@ -1,59 +1,59 @@
 #!/usr/bin/env python3
 
 import argparse
-import io
+from typing import Any, Optional
+
 import numpy as np
-from numpy.typing import ArrayLike, NDArray
-import tcod
-import PIL.Image as Image
-import PIL.ImageFont as ImageFont
-import PIL.ImageDraw as ImageDraw
-
-from tcod.loader import lib
-from tcod.tileset import CHARMAP_CP437
-
-CP437_LEN=256
-IMAGE_W_CELLS=16
-IMAGE_H_CELLS=16
-
-def write_cp437(font_path, width_px, height_px): # -> ndarray[uint8]
-    tileset = tcod.tileset.load_truetype_font(
-        font_path,
-        tile_width=width_px,
-        tile_height=height_px,
-    )
-    for i in range(256):
-        tileset.remap(tcod.tileset.CHARMAP_CP437[i], i)
-
-    buf = np.zeros(
-            shape=(IMAGE_W_CELLS, IMAGE_H_CELLS),
-            dtype=tcod.console.Console.DTYPE,
-            order="F"
-    )
-    for x in range(IMAGE_W_CELLS):
-        for y in range(IMAGE_H_CELLS):
-            buf[x, y] = tcod.tileset.CHARMAP_CP437[(IMAGE_W_CELLS * x) + y]
-
-    return buf
-
-def render(buf: NDArray[np.uint8], font: ImageFont, width_px: int, height_px: int):
-    canvas_w = IMAGE_W_CELLS * width_px
-    canvas_h = IMAGE_H_CELLS * height_px
-    canvas = Image.new('RGBA', (canvas_w, canvas_h))
+import tcod.tileset
+from numpy.typing import NDArray
+from PIL import Image, ImageDraw, ImageFont  # type: ignore
+
+# Code page 437 Unicode codepoints as a 16x16 grid, row-major order.
+CHARMAP_CP437 = np.asarray(tcod.tileset.CHARMAP_CP437, dtype=int).reshape(16, 16)
```

Comment:
lovely :)

Reply:
Yeah, I noticed that `buf` being the only thing returned from `write_cp437` was only being used to hold a grid of codepoints for the tileset to be rendered so I made this refactor.